src/components/BenefitPlanGroupBeneficiariesFilter.js

```diff
@@ -0,0 +1,70 @@
+import React from 'react';
+import _debounce from 'lodash/debounce';
+import { injectIntl } from 'react-intl';
+import { Grid } from '@material-ui/core';
+import { withTheme, withStyles } from '@material-ui/core/styles';
+import { TextInput, formatMessage } from '@openimis/fe-core';
+import { CONTAINS_LOOKUP, DEFAULT_DEBOUNCE_TIME } from '../constants';
+import { defaultFilterStyles } from '../util/styles';
+import BeneficiaryStatusPicker from '../pickers/BeneficiaryStatusPicker';
+
+function BenefitPlanBeneficiariesFilter({
+  intl, classes, filters, onChangeFilters, readOnly,
+}) {
+  const debouncedOnChangeFilters = _debounce(onChangeFilters, DEFAULT_DEBOUNCE_TIME);
+
+  const filterValue = (filterName) => filters?.[filterName]?.value;
+
+  const onChangeStringFilter = (filterName, lookup = null) => (value) => {
+    if (lookup) {
+      debouncedOnChangeFilters([
+        {
+          id: filterName,
+          value,
+          filter: `${filterName}_${lookup}: "${value}"`,
+        },
+      ]);
+    } else {
+      onChangeFilters([
+        {
+          id: filterName,
+          value,
+          filter: `${filterName}: "${value}"`,
+        },
+      ]);
+    }
+  };
+
+  return (
+    <Grid container className={classes.form}>
+      <Grid item xs={2} className={classes.item}>
+        <TextInput
+          module="socialProtection"
+          label="groupBeneficiary.id"
+          value={filterValue('group_Id')}
```

Comment:
If this is a TextInput, we should implement _filterTextFieldValue_ (const filterTextFieldValue = (filterName) => filters?.[filterName]?.value ?? EMPTY_STRING;) and use it to filter the 'group_Id'.

Reply:
done